diff --git a/bsp/lwip141/src/contrib/ports/xilinx/sys_arch.c b/bsp/lwip141/src/contrib/ports/xilinx/sys_arch.c
index 9d4e80f..262d436 100644
--- a/bsp/lwip141/src/contrib/ports/xilinx/sys_arch.c
+++ b/bsp/lwip141/src/contrib/ports/xilinx/sys_arch.c
@@ -557,7 +557,9 @@ portBASE_TYPE xHigherPriorityTaskWoken = pdFALSE;
 	if( xInsideISR != pdFALSE )
 	{
 		xReturn = xQueueSendFromISR( *pxMailBox, &pxMessageToPost, &xHigherPriorityTaskWoken );
-		portYIELD_FROM_ISR( xHigherPriorityTaskWoken );
+		if (xHigherPriorityTaskWoken == pdTRUE) {
+			portYIELD_FROM_ISR();
+		}
 	}
 	else
 	{
@@ -681,7 +683,9 @@ portBASE_TYPE xHigherPriorityTaskWoken = pdFALSE;
 	if( xInsideISR != pdFALSE )
 	{
 		lResult = xQueueReceiveFromISR( *pxMailBox, &( *ppvBuffer ), &xHigherPriorityTaskWoken );
-		portYIELD_FROM_ISR( xHigherPriorityTaskWoken );
+		if (xHigherPriorityTaskWoken == pdTRUE) {
+			portYIELD_FROM_ISR();
+		}
 	}
 	else
 	{
@@ -805,8 +809,7 @@ err_t sys_mutex_new( sys_mutex_t *pxMutex )
 {
 err_t xReturn = ERR_MEM;
 
-	//*pxMutex = xQueueCreateMutex();
-	*pxMutex = xSemaphoreCreateMutex();
+	*pxMutex = xQueueCreateMutex();
 
 	if( *pxMutex != NULL )
 	{
@@ -861,7 +864,9 @@ portBASE_TYPE xHigherPriorityTaskWoken = pdFALSE;
 	if( xInsideISR != pdFALSE )
 	{
 		xSemaphoreGiveFromISR( *pxSemaphore, &xHigherPriorityTaskWoken );
-		portYIELD_FROM_ISR( xHigherPriorityTaskWoken );
+		if (xHigherPriorityTaskWoken == pdTRUE) {
+			portYIELD_FROM_ISR();
+		}
 	}
 	else
 	{
